src/core/structures/context.ts

```diff
@@ -31,53 +31,73 @@ export class Context extends CoreContext<Message, ChatInputCommandInteraction> {
     }
 
     public get id(): Snowflake {
-        return this.ctx.val.id;
+        return safeUnwrap(this.ctx
+                            .map(m => m.id)
+                            .mapErr(i => i.id));
     }
 
     public get channel() {
```

Comment:
Why this isn't typed while almost every other one does

Reply:
it is inferred. Im not touching the return types of context until we decide what to do with sern cord